### PR TITLE
RemoveEmptyJavaDocParameters: added tests + fixed bugs

### PR DESCRIPTION
**Pull request by joanvr**
## What's changed?
There was a lot of missing scenarios on the tests that were causing bugs (wrong format of last lines, first lines, even index out of bounds on some very rare scenarios).
Added all the missing combinations of [param, return, throws], [regular, singleLine, firstLine], [withSpace, noSpace].
Fixed the bugs not originally covered with the new tests.

## What's your motivation?
Fixes https://github.com/moderneinc/support-app/issues/14 and hopefully all other bugs in this recipe.

## Anyone you would like to review specifically?
@timtebeek @knutwannheden 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/staticanalysis/RemoveEmptyJavaDocParameters.java [hunk 1]
```diff
@@ -79,6 +79,12 @@ public Javadoc visitDocComment(Javadoc.DocComment javadoc, ExecutionContext ctx)
                     boolean useNewBody = false;
 
                     List<Javadoc> body = javadoc.getBody();
+                    // To fix elements on the first line without space
+                    if (!body.isEmpty() && !(body.get(0) instanceof Javadoc.LineBreak)) {
+                        body = new ArrayList<>(body);
+                        body.add(0, null); // we can use null since this element is never going to be used.
+                    }
```
Comment (joanvr) [inline src/main/java/org/openrewrite/staticanalysis/RemoveEmptyJavaDocParameters.java:86]: by adding this trailing element, we can always process the elements in this loop, and we no longer require the specific element visitors, that were causing a bit of confusion and complexity to debug the recipe. Now those are gone and changed to checking boolean methods (isEmpty...)

**Review by timtebeek — APPROVED** [review]
Nice! I'd briefly explored this previously, but indeed found the code hard to follow. Glad that's sorted now.
